utils/text_test.go

```diff
@@ -72,12 +72,57 @@ func TestTokenizeStringByChars(t *testing.T) {
 	}
 }
 
+func TestTokenizeStringByUnicodeSeg(t *testing.T) {
+	tokenizerTests := []struct {
+		text   string
+		result []string
+	}{
+		// test cases taken from Apache Lucene (TestStandardAnalyzer.java) to assert equivalency
```

Comment:
Our ElasticSearch implementation of name searching uses the `StandardTokenizer` from Apache Lucene (https://github.com/apache/lucene-solr/blob/1d85cd783863f75cea133fb9c452302214165a4d/lucene/core/src/java/org/apache/lucene/analysis/standard/StandardTokenizerImpl.java) so added some of it's unit tests here to see how close we are.